js/web/components/Profile.js

```diff
@@ -0,0 +1,377 @@
+        ),
+      ]
+    }
+  }, [profilePublishedReleases])
+
+  useEffect(() => {
+    const getUserData = async (profilePubkey) => {
+      await getHubsForUser(profilePubkey)
+      const [collectionIds, publishedIds] = await getUserCollectionAndPublished(
+        profilePubkey
+      )
+      setProfileCollectionIds(collectionIds)
+    }
+    if (profilePubkey) {
+      setFetchedUser(true)
+      getUserData(profilePubkey)
+    }
+  }, [profilePubkey])
+
+  useEffect(() => {
+    let viewIndex
+    let updatedView = views.slice()
+    if (profilePubkey) {
+      const releases = filterReleasesPublishedByUser(profilePubkey)
+      setProfilePublishedReleases(releases)
+      viewIndex = updatedView.findIndex((view) => view.name === 'releases')
+      updatedView[viewIndex].playlist = releases
+      setFetchedReleases(true)
+      setProfileCollectionReleases(filterReleasesList(profileCollectionIds))
+      viewIndex = updatedView.findIndex((view) => view.name === 'collection')
+      updatedView[viewIndex].playlist = filterReleasesList(profileCollectionIds)
+      setFetchedCollection(true)
+      const hubs = filterHubsForUser(profilePubkey)
+      setProfileHubs(hubs)
+      setFetchedHubs(true)
+    }
+    setViews(updatedView)
+  }, [profileCollectionIds, releaseState])
+
+  useEffect(() => {
+    if (fetchedReleases && profilePublishedReleases?.length > 0) {
+      setActiveView(0)
+    }
+    if (
+      fetchedCollection &&
+      fetchedReleases &&
+      profilePublishedReleases?.length === 0 &&
+      profileCollectionReleases?.length > 0
+    ) {
+      setActiveView(1)
+    }
+    if (
+      fetchedCollection &&
+      fetchedReleases &&
+      fetchedHubs &&
+      profilePublishedReleases?.length === 0 &&
+      (profileCollectionReleases?.length === 0) & (profileHubs?.length > 0)
+    ) {
+      setActiveView(2)
+    }
+  }, [profilePublishedReleases, profileCollectionReleases])
+
+  useEffect(() => {
+    let viewIndex
+    let updatedView = views.slice()
+
+    if (profilePublishedReleases?.length > 0) {
+      viewIndex = updatedView.findIndex((view) => view.name === 'releases')
+
+      updatedView[viewIndex].visible = true
+    }
+    if (profileCollectionReleases?.length > 0) {
+      viewIndex = updatedView.findIndex((view) => view.name === 'collection')
+      updatedView[viewIndex].visible = true
+    }
+
+    setViews(updatedView)
+  }, [profilePublishedReleases, profileCollectionReleases])
+
+  useEffect(() => {
+    let viewIndex
+    let updatedView = views.slice()
+
+    if (profileHubs && profileHubs?.length > 0 && fetchedHubs) {
+      viewIndex = updatedView.findIndex((view) => view.name === 'hubs')
+      updatedView[viewIndex].visible = true
+    }
+    setViews(updatedView)
+  }, [profileHubs])
+
+  const viewHandler = (event) => {
+    const index = parseInt(event.target.id)
+    setActiveView(index)
+  }
+
+  const fetchedAll = fetchedCollection && fetchedHubs && fetchedReleases
+
+  const noProfile =
```

Comment:
same comment as on hubs, this should be in a `useEffect`  / more closely associated to the states its working w